app/controllers/api/configuration_script_sources_controller.rb

```diff
@@ -16,20 +16,15 @@ def delete_resource_main_action(type, config_script_src, _data)
       {:task_id => config_script_src.delete_in_provider_queue}
     end
 
-    def create_resource(_type, _id, data)
-      validate_attrs(data)
+    def create_resource(type, _id, data)
+      # TODO: convert configuration script api to (also?) support standard `ems_id` key
       manager_id = parse_id(data['manager_resource'], :providers)
-      raise 'Must specify a valid manager_resource href or id' unless manager_id
-      manager = resource_search(manager_id, :providers)
+      create_ems_resource(type, {'ems_id' => manager_id, 'name' => data['name']}) do |manager, klass|
+        # TODO: introduce supports for configuration scripts create
+        raise "ConfigurationScriptSource cannot be added to #{model_ident(manager, :provider)}" unless klass.respond_to?(:create_in_provider_queue)
```

Comment:
Don't want to hold this PR up for this, but we should add `supports :create` to any types which support this rather than checking for the method existence.

Reply:
Looks like there are only 2 places where we define this method this method. Should not be too hard to get the `supports :creates` into those spots

Reply:
Okay I think everything is merged to allow you to do `klass.supports?(:create)` here